Cargo.toml

```diff
@@ -184,3 +184,71 @@ required-features = ["rustls", "ws"]
 name = "subscribe_contract_events_with_meta"
 path = "examples/subscribe_contract_events_with_meta.rs"
 required-features = ["rustls", "ws"]
+
+[[example]]
```

Comment:
Can we avoid these explicit declarations somehow?

Reply:
Hi @gakonst, thanks for pointing this out. 

Having a cargo workspace makes it more complicated, since nested example files are not resolved automatically.

We could provide a `Cargo.toml` in the examples directory (also done by the [tokio](https://github.com/tokio-rs/tokio/tree/master/examples) team). In this case `examples` would become a new workspace member. That would offload the root `Cargo.toml`, at the cost of maintaining a new workspace member.

Can it work for you?  Any other advice is welcome.

Thank you

Reply:
Yep - I like doing that that way.

Reply:
Since examples are structured in folders, we can create a standalone crate for each. We then use the native cargo examples layout:

```bash
examples
├── big-numbers
│   ├── Cargo.toml
│   ├── README.md
│   └── examples
│       ├── bn_comparison_equivalence.rs
│       ├── bn_conversion.rs
│       ├── bn_create_instances.rs
│       ├── bn_math_operations.rs
│       └── bn_utilities.rs
├── contracts
│   ├── Cargo.toml
│   ├── README.md
│   └── examples
│       ├── IERC20.json
│       ├── contract_abigen.rs
│       ├── contract_events.rs
│       ├── contract_instances.rs
│       └── contract_methods.rs
```

We can run examples with this
`cargo run -p <example-crate-name> --example <example-name>`
i.e.
`cargo run -p examples-contracts --example contract_abigen`

Can it work for you?

Reply:
I restructured all examples with this approach:
* All files have been moved to their espective crate
* Examples can be run with `cargo run -p <example-crate-name> --example <example-name>`
* Build succeeds (locally)
* CI is broken, need to fix it 😆 

Can you guys take a look? @gakonst @DaniPopes

Reply:
Given they're grouped I am OK doing it that way